main/src/ca/uwaterloo/flix/language/phase/Parser2.scala

```diff
@@ -2468,8 +2468,24 @@ object Parser2 {
       expect(TokenKind.KeywordNew, SyntacticContext.Expr.OtherExpr)
       Type.ttype()
 
-      // NewObject or InvokeConstructor?
-      if (at(TokenKind.CurlyL)) {
+      // NewStruct, NewObject, or InvokeConstructor?
+      if (at(TokenKind.CurlyL) && (nth(1) == TokenKind.NameLowerCase || (nth(1) == TokenKind.CurlyR && nth(2) == TokenKind.At))) {
```

Comment:
I am surprised this works. Will it not shadow the next case, so to speak?

Reply:
The cases are
```
new StructName { } @ r // case 1
new StructName {field1 = val, ...} @ r // case 2
new Type { } // case 3
new Type { def fn1 ... } // case 4
new Type( ... ) // case 5
```
My cases don't shadow the others because
- case 1 checks for the `@` in the empty struct case to avoid conflict with case 3
- case 2 checks for a lowercase name to avoid with case 4(which must have a `def` instead
- case 1 and 2 check for `{` to prevent conflict with case 5

Reply:
OK- that helps!